core/fullHistory/historyProcessor.go

```diff
@@ -82,54 +83,69 @@ func (hp *historyProcessor) PutTransactionsData(historyTxsData *HistoryTransacti
 
 	epoch := historyTxsData.HeaderHandler.GetEpoch()
 
-	err := hp.hashEpochProc.SaveEpoch(historyTxsData.HeaderHash, epoch)
+	err := hp.hashEpochStorer.SaveEpoch(historyTxsData.HeaderHash, epoch)
 	if err != nil {
-		log.Warn("epochHashProcessor:cannot save header hash", "error", err.Error())
+		log.Warn("epochHashProcessor: cannot save header hash", "error", err.Error())
 		return err
 	}
 
 	for _, mb := range body.MiniBlocks {
-		mbHash, err := core.CalculateHash(hp.marshalizer, hp.hasher, mb)
+		err = hp.saveMiniblockData(historyTxsData, mb, epoch)
 		if err != nil {
 			continue
 		}
+	}
 
-		err = hp.hashEpochProc.SaveEpoch(mbHash, epoch)
-		if err != nil {
-			log.Warn("epochHashProcessor:cannot save miniblock hash", "error", err.Error())
-			continue
-		}
+	return nil
+}
 
-		var status core.TransactionStatus
-		if mb.ReceiverShardID == hp.selfShardID {
-			status = core.TxStatusExecuted
-		} else {
-			status = core.TxStatusPartiallyExecuted
-		}
+func (hp *historyProcessor) saveMiniblockData(historyTxsData *HistoryTransactionsData, mb *block.MiniBlock, epoch uint32) error {
+	mbHash, err := core.CalculateHash(hp.marshalizer, hp.hasher, mb)
+	if err != nil {
+		return err
+	}
 
-		historyTx := buildTransaction(historyTxsData.HeaderHandler, historyTxsData.HeaderHash, mbHash, mb.ReceiverShardID, mb.SenderShardID, status)
+	err = hp.hashEpochStorer.SaveEpoch(mbHash, epoch)
+	if err != nil {
+		log.Warn("epochHashProcessor: cannot save miniblock hash", "error", err.Error())
+		return err
+	}
+
+	var status core.TransactionStatus
+	if mb.ReceiverShardID == hp.selfShardID {
+		status = core.TxStatusExecuted
+	} else {
+		status = core.TxStatusPartiallyExecuted
+	}
 
-		historyTxBytes, err := hp.marshalizer.Marshal(historyTx)
+	historyTx := buildTransaction(historyTxsData.HeaderHandler, historyTxsData.HeaderHash, mbHash, mb.ReceiverShardID, mb.SenderShardID, status)
```

Comment:
`historyTxs` (plural)?

Reply:
it's a little weirder because it's a single transaction that's the same for all transactions in a miniblock.... I can renamed it if you strongly suggest